### PR TITLE
[iOS]Cache images when loading floor map

### LINKED ISSUE #770: [iOS]Cache images when loading floor map
## Kind (Required)

- Improvement

 ## Overview (Required)

floor map is a page that you will see many times.
So, cache images when loaded.

 ## Links	

- [Nuke](https://github.com/kean/Nuke)

### PR DESCRIPTION
**Pull request by tomu28**
## Issue
- close #770

## Overview (Required)
- Cache images when loading floor map

## Links
- [Nuke - LRU Memory Cache](https://github.com/kean/Nuke/tree/master#lru-memory-cache)

## Screenshot
No UI changes.

Before | After
:--: | :--:
 |

### CODE REVIEW

File: ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift [hunk 1]
```diff
@@ -29,19 +29,39 @@ final class FloorMapViewController: ContentViewController {
...
+    
+    /// This method caches the floor map image in LRU memory when loading from server.
     func loadMap() {
-        let width = imageView.image?.size.width
-        let height = imageView.image?.size.height
-
+        // Configure cache
+        ImageCache.shared.costLimit = 1024 * 1024 * 100 // 100 MB
+        ImageCache.shared.countLimit = 100
+        ImageCache.shared.ttl = 120 // Invalidate image after 120 sec
```
Comment (tomu28) [inline ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift]: During [the event](https://droidkaigi.jp/2020/en/) of the day,
I think that I look at the timeline and check the floor map many times in a short time.
So, I adjusted the cache time to 120 seconds.
If you have any other ideas, please let me know!😊

File: ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift [hunk 2]
```diff
@@ -29,19 +29,39 @@ final class FloorMapViewController: ContentViewController {
...
+        
+        ImageLoadingOptions.shared.failureImage = UIImage(named: "map") // The set image is applied when reading from the server fails
```
Comment (ry-itto) [inline ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift]: Please use `Asset.map.image` 🙏

File: ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift (hunk 1 above)
Comment (ry-itto) [reply ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift]: I think it is ok. 😄
But I think `costLimit` and `countLimit` are too large value for this. 😭

**Comment by ry-itto** [comment]
How can I confirm images are cached?
If you know, please tell me. 🙏

**Comment by tomu28** [comment]
Thank you for the review😄
I will fix it once confirmed!

**Comment by tomu28** [comment]
@ry-itto 
>How can I confirm images are cached?
If you know, please tell me. 🙏

Oh, sorry!
The explanation was insufficient🙇‍♂️

You can check that the image is cached by following the steps below!
Thank you for your help in advance.😄

## Step1:
Until the image from the server is loaded, the default map image specified in `Storyboard` is displayed.
So, change the set image(`map.png`) to make it easier to confirm that the image is loaded from the server.

### Reference image:
This time, I changed from `map` to `header_about`.

refarenceImage

## Step2:
[When successfully loaded,](https://github.com/DroidKaigi/conference-app-2020/pull/775/files#diff-4bb5ebce785a29b29305cf99ad3c3338R57-R59) showing cached images.
Please run and check🙏
At first, `header_about` is displayed, but it will switch when the image is loaded.

## Step3:

1. For 120 seconds, the floor map image is cached.(`ImageCache.shared [request]` contains image data as cache) So, open another screen such as the timeline once.
1. Open the floor map again.
1. [If there is an image already cached, display that image.](https://github.com/DroidKaigi/conference-app-2020/pull/775/files#diff-4bb5ebce785a29b29305cf99ad3c3338R52-R53) At that time, you can see that [`ImagePipeline.shared.loadImage`](https://github.com/DroidKaigi/conference-app-2020/pull/775/files#diff-4bb5ebce785a29b29305cf99ad3c3338R55) is not called.

If you use breakpoint or Step Over, it will be easier to confirm.
[In this part!](https://github.com/DroidKaigi/conference-app-2020/pull/775/files#diff-4bb5ebce785a29b29305cf99ad3c3338R52-R61)

Finally, If there is anything I can help you with, please do not hesitate to reply me!

File: ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift [hunk 3]
```diff
@@ -32,8 +32,8 @@ private extension FloorMapViewController {
     /// This method caches the floor map image in LRU memory when loading from server.
     func loadMap() {
         // Configure cache
-        ImageCache.shared.costLimit = 1024 * 1024 * 5 // 5 MB
-        ImageCache.shared.countLimit = 10
+        ImageCache.shared.costLimit = 1024 * 1024 * 50 // 50 MB
```
Comment (tomu28) [inline ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift]: The problem that operation becomes unstable when the value of `costLimit` and `countLimit ` is as follows was confirmed.

```swift
ImageCache.shared.costLimit = 1024 * 1024 * 5 // 5 MB
ImageCache.shared.countLimit = 10
```

So, I revised the value again.
However, I’m  not sure if this value is appropriate.
I'll find out which value is right for [this part](https://github.com/DroidKaigi/conference-app-2020/pull/775/files#diff-4bb5ebce785a29b29305cf99ad3c3338R35-R36), but I'd be grateful if you could give me an opinion😄

**Pull request merged**